compiler/sketch-library/lib/index.js

```diff
@@ -86,30 +86,64 @@ module.exports = function generateSketchLibrary(
     // TODO: remove previous export
   }
 
-  return new Promise((resolve, reject) => {
-    log(`Getting the configuration of the workspace`)
-    exec(`node "${compiler}" config "${workspace}"`, (err, stdout, stderr) => {
-      if (err) {
-        err.stdout = stdout
-        err.stderr = stderr
-        return reject(err)
-      }
+  return Promise.all([
+    new Promise((resolve, reject) => {
+      log(`Getting the tokens of the workspace`)
+      exec(
+        `node "${compiler}" flatten --workspace "${workspace}"`,
+        (err, stdout, stderr) => {
+          if (err) {
+            err.stdout = stdout
+            err.stderr = stderr
+            return reject(err)
+          }
 
-      return resolve(stdout)
-    })
-  })
-    .then(rawConfig => {
-      const compilerConfig = JSON.parse(rawConfig)
+          return resolve(JSON.parse(stdout))
+        }
+      )
+    }),
+    new Promise((resolve, reject) => {
+      exec(
+        `node "${compiler}" config "${workspace}"`,
+        (err, stdout, stderr) => {
+          if (err) {
+            err.stdout = stdout
+            err.stderr = stderr
+            return reject(err)
+          }
 
+          return resolve(JSON.parse(stdout))
+        }
+      )
+    }),
+  ])
+    .then(([tokens, config]) => {
       return {
         paths: {
           output: babelOutput,
           sketchFile: sketchFilePath,
-          workspace: compilerConfig.paths.workspace,
-          colors: compilerConfig.paths.colors,
-          textStyles: compilerConfig.paths.textStyles,
-          components: compilerConfig.paths.components,
+          workspace,
+          components: config.paths.components,
         },
+        tokens: tokens.files.reduce(
+          (prev, file) => {
+            file.contents.value.forEach(token => {
```

Comment:
It might be worth checking that `file.contents.type` is `"flatTokens"` to make this more resilient in case we end up including other kinds of converted content (SVGs or something?) in the flattened JSON

https://github.com/Lona/flatjson/blob/dad18051719477e80254d0044ccf2f5e45eb8410/flat-token-types.ts#L42

Reply:
Good idea 👍